src/acl.c

```diff
@@ -1531,56 +1531,28 @@ static int ACLSelectorCheckKey(aclSelector *selector, const char *key, int keyle
     return ACL_DENIED_KEY;
 }
 
-/* Returns if a given command may possibly access channels. For this context,
- * the unsubscribe commands do not have channels. */
-static int ACLDoesCommandHaveChannels(struct redisCommand *cmd) {
-    return (cmd->proc == publishCommand
-        || cmd->proc == subscribeCommand
-        || cmd->proc == psubscribeCommand
-        || cmd->proc == spublishCommand
-        || cmd->proc == ssubscribeCommand);
-}
-
-/* Checks a channel against a provide list of channels. */
-static int ACLCheckChannelAgainstList(list *reference, const char *channel, int channellen, int literal) {
+/* Checks a channel against a provided list of channels. The is_pattern 
+ * argument controls whether the input channel is evaluated as a channel pattern
+ * (like in PSUBSCRIBE) or a single channel (like in SUBSCRIBE). */
```

Comment:
It's good that now the "pattern" flag is passed all the way down and isn't being exchanged with a "literal" argument, so the comment explaining it needs to be put in just one place.
but i still had hard time wrapping my head around it.
eventually when i did, i tried to improve the comment for the next time i forget it.
hope it works.
```suggestion
/* Checks a channel against a provided list of channels. The is_pattern 
 * argument should only be used when subscribing (not when publishing)
 * and controls whether the input channel is evaluated as a channel pattern
 * (like in PSUBSCRIBE) or a single channel (like in SUBSCRIBE). 
 * Note that a plan channel name like in PUBLISH or SUBSCRIBE can be
 * matched against ACL channel patterns, but the pattern provided in PSUBSCRIBE
 * can only be matched as literal against an ACL pattern (using plain string compare). */
```